app/grandchallenge/algorithms/models.py

```diff
@@ -69,6 +69,38 @@
 JINJA_ENGINE = sandbox.ImmutableSandboxedEnvironment()
 
 
+class AlgorithmInterfaceManager(models.Manager):
+    def get_existing_interface_for_inputs_and_outputs(
+        self, *, inputs, outputs
+    ):
+        try:
+            return self.annotate(
+                input_count=Count("inputs", distinct=True),
+                output_count=Count("outputs", distinct=True),
+            ).get(
+                inputs__in=inputs,
+                outputs__in=outputs,
+                input_count=len(inputs),
+                output_count=len(outputs),
+            )
+        except ObjectDoesNotExist:
+            return None
+
+
+class AlgorithmInterface(models.Model):
```

Comment:
We probably should use a `UUIDModel` as we will likely need to serialize these at some point.

Reply:
Hadn't thought of that, thanks!